### PR TITLE
Refactor: change edit page to use hooks

### PR DESCRIPTION
**Pull request by alexanderleegs**
This PR refactors editPage to use hooks, in preparation to store site colours using context instead. In addition, this PR also removes an extra log statement in Workspace.

### CODE REVIEW

File: src/layouts/EditPage.jsx [hunk 1]
```diff
@@ -100,165 +99,160 @@ const getBackButtonInfo = (resourceCategory, collectionName, siteName) => {
...
-
       // here, we need to re-add the front matter of the markdown file
       const upload = concatFrontMatterMdBody(frontMatter, editorValue);
 
       // encode to Base64 for github
       const base64Content = Base64.encode(upload);
       const params = {
         content: base64Content,
-        sha: state.sha,
+        sha: sha,
```
Comment (kwajiehao) [inline src/layouts/EditPage.jsx:253]: nit: can just write `sha,` instead of `sha: sha`

**Review by kwajiehao — APPROVED** [review]
lgtm, left a couple of minor nits

File: src/layouts/EditPage.jsx [hunk 2]
```diff
@@ -384,197 +348,173 @@ export default class EditPage extends Component {
          */
 
       const fileData = fileReader.result.split(',')[1];
-      this.stageFileForUpload(fileName, fileData);
+      stageFileForUpload(fileName, fileData);
     });
     fileReader.readAsDataURL(event.target.files[0]);
-    this.toggleImageModal()
+    setIsSelectingImage(!isSelectingImage)
```
[the remark this answers is not in the document]
Comment (alexanderleegs) [reply src/layouts/EditPage.jsx]: Good catch, resolved in https://github.com/isomerpages/isomercms-frontend/pull/337/commits/8b92bbe7ae9c36d385305d5eb2322c9b9293727d

File: src/layouts/EditPage.jsx (hunk 1 above)
Comment (alexanderleegs) [reply src/layouts/EditPage.jsx]: Resolved in https://github.com/isomerpages/isomercms-frontend/pull/337/commits/0b82a283d15f16cfaaafb0ffa5251f093030137d

**Pull request merged**